src/util/namespace.h

```diff
@@ -84,8 +84,14 @@ class namespacet:public namespace_baset
 
   using namespace_baset::lookup;
 
-  // these do the actual lookup
+
+  /// Searches for a symbol named \p name in `symbol_table1` and returns a
+  /// pointer to it in output parameter \p symbol when found. When no symbol is
+  /// found, it performs the same lookup in `symbol_table2`.
```

Comment:
See my comment elsewhere: work backwards by first considering who the audience is, and what you want them to know. To me, the proposed comment says that `lookup` does a search. I'd rather not have that comment, to be honest, as it doesn't actually tell me much.

Add comments where they add value, don't comment for the comment's sake when the name is informative.